lib/matplotlib/tests/test_axes.py

```diff
@@ -4335,6 +4335,17 @@ def test_pandas_indexing_hist():
     fig, axes = plt.subplots()
     axes.hist(ser_2)
 
+@cleanup
+def test_axis_set_tick_params_labelsize_labelcolor():
+    # Tests fix for issue 4346
+    axis_1 = plt.subplot()
+    axis_1.yaxis.set_tick_params(labelsize=30, labelcolor='red', direction='out')
+
+    # Expected values after setting the ticks
+    assert axis_1.yaxis.majorTicks[0]._size == 4.0
```

Comment:
The reason for asking that is that, long term, we reserve the right to change the private API with no warning or deprecation cycle.  Tests that only use public API give confidence that we have done that right but that is eroded if we have to touch the tests as part of the refactor as well.